snuba/api.py

```diff
@@ -183,12 +183,14 @@ def parse_request_body(request):
 
 def validate_request_content(body, schema, timer):
     try:
-        schemas.validate(body, schema)
+        request = schema.validate(body)
     except jsonschema.ValidationError as error:
         raise BadRequest(str(error)) from error
 
     timer.mark('validate_schema')
 
+    return {**request.body}
```

Comment:
I assume you are creating a new dictionary here in order to have a copy. Is that correct? If yes, I think we should think about a plan for mutations of the body and what we want to allow. My opinion is just returning a frozenset or any immutable mapping and completely disallow mutations of the body of the request. I am suggesting this because we are giving more structure to this code than it was used to have, thus keeping track of mutations to a shared daat structure becomes harder.

Reply:
> I assume you are creating a new dictionary here in order to have a copy. Is that correct?

Yeah, this is just for change minimization with the existing code.

> If yes, I think we should think about a plan for mutations of the body and what we want to allow. My opinion is just returning a frozenset or any immutable mapping and completely disallow mutations of the body of the request. I am suggesting this because we are giving more structure to this code than it was used to have, thus keeping track of mutations to a shared daat structure becomes harder.

I generally agree your viewpoint here. I think typing will help here since we can enforce this by using `Mapping` rather than `MutableMapping`. I don't think we should try and address this as part of this change though, but keep it in mind moving forward.

Reply:
mind filing a task to rely on immutable data if you do not plan to address the problem right away ?